src/java/fluree/db/SID.java

```diff
@@ -3,33 +3,25 @@
 import java.lang.Comparable;
 
 public final class SID implements Comparable {
-    final int namespaceCode;
-    final long[] nameCodes;
+    public final int namespaceCode;
+    public final long[] nameCodes;
 
-    public SID(int nsCode, long... nameCodes) {
+    public SID(int nsCode, long[] nameCodes) {
         this.namespaceCode = nsCode;
         this.nameCodes = nameCodes;
     }
 
-    public int getNamespaceCode() {
```

Comment:
Why did the getters need to be removed?

Reply:
We need to add them back. This is a Java class, not Clojure. Those fields should not be public, and they should not be settable. This change makes them settable. 

I don't think anything about the Java class needed to change to fix this bug.

Reply:
My understanding is that `final` indicates that the field cannot be mutated and is more than sufficient for safety, but I've gone ahead and reverted the changes.

Reply:
I think we should strive to write code as idiomatically as possible in whatever language we're writing in. I would not have written a clojure namespace hiding data behind a function, but that is the Java way, and this is a Java class.

Reply:
My bold take is that idiomatic java is in this case just wrong - getters are a bad design and the convention for using them comes from a time when `final` did not exist. 

Especially since this is in performance sensitive code my thought is why waste a stack frame on something we don't need. Same thought with changing the signature from `long...` to `long[]`, we only ever supply an array so no need to generate vararg handling. But in the end it's only a stack frame and I'm sure the JIT will handle both cases correctly anyways.